### ISSUE TITLE
plugin: types not in sync with code

### ISSUE THREAD

**Issue by susnux** [eslint-stylistic/eslint-stylistic#762]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

The types in the published packages are not in sync with the JavaScript code and also do not match the types in this repository (dts/index).

The plugin only exports a default export, but the shipped types in the package (4.2.0) state:

```ts
export { type Configs, type Rules, type StylisticCustomizeOptions, UnprefixedRuleOptions, configs, customize, plugin as default };
```

So I tried to import `configs` or `customize` as named imports, but obviously failed as they are not exported in the JavaScript code.

### Reproduction

https://stackblitz.com/edit/github-y5xx7akw?file=README.md

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by manferlo81** [comment]
Same here
